### PR TITLE
multiview pca refactor

### PR DESCRIPTION
**Pull request by themattinthehatt**

### CODE REVIEW

**Review by danbider — CHANGES_REQUESTED** [review]
looks great; minor `datamodules.py` suggestions tangential to your changes

File: pose_est_nets/datasets/datamodules.py [hunk 1]
```diff
@@ -4,16 +4,14 @@
...
 from typeguard import typechecked
 from typing import Literal, List, Optional, Tuple, Union
 
 from pose_est_nets.datasets.dali import video_pipe, LightningWrapper
 from pose_est_nets.datasets.datasets import BaseTrackingDataset, HeatmapDataset
-from pose_est_nets.datasets.utils import clean_any_nans, split_sizes_from_probabilities
-from pose_est_nets.utils.heatmap_tracker_utils import format_mouse_data
+from pose_est_nets.datasets.utils import split_sizes_from_probabilities
 
 _TORCH_DEVICE = "cuda" if torch.cuda.is_available() else "cpu"
```
[the remark this answers is not in the document]
Comment (themattinthehatt) [reply pose_est_nets/datasets/datamodules.py:16]: not sure I understand - what's the worry here?

File: pose_est_nets/datasets/datamodules.py [hunk 2]
```diff
@@ -289,88 +299,6 @@ def setup_unlabeled(self):
...
-            proj_discarded.numpy(), empirical_epsilon_percentile, axis=0
-        )
-        print(epsilon)
-        self.loss_param_dict["pca"]["epsilon"] = torch.tensor(
-            epsilon,
-            dtype=torch.float32,
-            device=_TORCH_DEVICE,  # TODO: be careful for multinode
-        )
-
     def unlabeled_dataloader(self):
```
[the remark this answers is not in the document]
Comment (themattinthehatt) [reply pose_est_nets/datasets/datamodules.py:302]: yep I can look into this

File: pose_est_nets/datasets/datamodules.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (danbider) [reply pose_est_nets/datasets/datamodules.py:16]: just a slight worry that `_TORCH_DEVICE` is hard-coded, somehow interfering with any other config stuff.

[the remark this answers is not in the document]
Comment (themattinthehatt) [reply pose_est_nets/datasets/datamodules.py:16]: oh nevermind I understand - yeah that could be an issue; but we have many other issues to solve before getting there

File: pose_est_nets/datasets/datamodules.py [hunk 3]
```diff
@@ -289,88 +299,6 @@ def setup_unlabeled(self):
             auto_reset=True,  # TODO: verify what "reseting" means
         )
 
-    # TODO: could be separated from this class
```
[the remark this answers is not in the document]
Comment (themattinthehatt) [reply pose_est_nets/datasets/datamodules.py:292]: :)

File: pose_est_nets/datasets/datamodules.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (themattinthehatt) [reply pose_est_nets/datasets/datamodules.py:302]: good suggestion, much cleaner now

**Review by danbider — APPROVED** [review]

**Pull request merged**